views/partials/_toc-patterns.njk

```diff
@@ -1,43 +1,22 @@
-      <a href="/patterns/check-a-service-is-suitable/">Check a service is suitable</a>
-    </li>
-    <li class="app-c-toc__section-item{% if path == 'patterns/check-answers' %} app-c-toc__section-item--current{% endif %}">
-      <a href="/patterns/check-answers/">Check answers</a>
-    </li>
-    <li class="app-c-toc__section-item{% if path == 'patterns/confirm-an-email-address' %} app-c-toc__section-item--current{% endif %}">
-      <a href="/patterns/confirm-an-email-address/">Confirm an email address</a>
-    </li>
-    <li class="app-c-toc__section-item{% if path == 'patterns/confirmation-pages' %} app-c-toc__section-item--current{% endif %}">
-      <a href="/patterns/confirmation-pages/">Confirmation pages</a>
-    </li>
-    <li class="app-c-toc__section-item{% if path == 'patterns/create-a-username' %} app-c-toc__section-item--current{% endif %}">
-      <a href="/patterns/create-a-username/">Create a username</a>
-    </li>
-    <li class="app-c-toc__section-item{% if path == 'patterns/create-accounts' %} app-c-toc__section-item--current{% endif %}">
-      <a href="/patterns/create-accounts/">Create accounts</a>
-    </li>
-    <li class="app-c-toc__section-item{% if path == 'patterns/email-addresses' %} app-c-toc__section-item--current{% endif %}">
-      <a href="/patterns/email-addresses/">Email addresses</a>
-    </li>
-    <li class="app-c-toc__section-item{% if path == 'patterns/fill-in-a-form' %} app-c-toc__section-item--current{% endif %}">
-      <a href="/patterns/fill-in-a-form/">Fill in a form</a>
-    </li>
-    <li class="app-c-toc__section-item{% if path == 'patterns/names' %} app-c-toc__section-item--current{% endif %}">
-      <a href="/patterns/names/">Names</a>
-    </li>
-    <li class="app-c-toc__section-item{% if path == 'patterns/national-insurance-numbers' %} app-c-toc__section-item--current{% endif %}">
-      <a href="/patterns/national-insurance-numbers/">National Insurance numbers</a>
-    </li>
-    <li class="app-c-toc__section-item{% if path == 'patterns/passwords' %} app-c-toc__section-item--current{% endif %}">
-      <a href="/patterns/passwords/">Passwords</a>
-    </li>
-    <li class="app-c-toc__section-item{% if path == 'patterns/question-pages' %} app-c-toc__section-item--current{% endif %}">
-      <a href="/patterns/question-pages/">Question pages</a>
-    </li>
+
+    {% for item in navigation %}
+      {% if item.url in path %}
+
+      {% set items = item.items or [] %}
+      {% for theme, items in items | groupby("theme") %}
+        {% if theme != 'undefined' %}
+          <h4 class="govuk-caption-m">{{ theme }}</h4>
+        {% endif %}
+        {% for subitem in items %}
+          <li class="app-c-toc__section-item{% if subitem.url == path %} app-c-toc__section-item--current{% endif %}">
+            <a href="/{{ subitem.url }}">{{ subitem.label }}</a>
```

Comment:
We're currently having to include `/` wherever we're linking in the navigation, because the paths built by the navigation plugin are relative but don't include the leading slash. Would it make more sense to include the leading slash when generating the URLs in the plugin instead?

Reply:
Tried to include them, but unfortunately if we're building the urls with a leading `/` we won't be able to verify them in `path` (e.g. `if item.url in path`)